src/drag-drop-polyfill-scroll-behaviour.ts

```diff
@@ -1,368 +1,376 @@
-        else if( scrollIntention === ScrollIntention.LEFT_OR_TOP ) {
+    const currentScrollOffset = (axis === ScrollAxis.HORIZONTAL) ? (scrollBounds.scrollX) : (scrollBounds.scrollY);
 
-            // is already at left/top edge
-            return (currentScrollOffset <= 0);
-        }
-        // no scroll
-        return true;
-    }
+    // wants to scroll to the right/bottom
+    if( scrollIntention === ScrollIntention.RIGHT_OR_BOTTOM ) {
 
-    //</editor-fold>
+        const maxScrollOffset = (axis === ScrollAxis.HORIZONTAL) ? ( scrollBounds.scrollWidth - scrollBounds.width ) : ( scrollBounds.scrollHeight -
+                                                                                                                         scrollBounds.height );
 
-    //<editor-fold desc="public api">
+        // is already at the right/bottom edge
+        return currentScrollOffset >= maxScrollOffset;
+    }
+    // wants to scroll to the left/top
+    else if( scrollIntention === ScrollIntention.LEFT_OR_TOP ) {
 
-    export interface ScrollOptions {
-        // threshold in px. when distance between scrollable element edge and touch position is smaller start programmatic scroll.
-        // defaults to 75px
-        threshold?:number;
-        // function to customize the scroll velocity
-        // velocity param: distance to scrollable element edge
-        // threshold: the threshold used to determine when scrolling should start
-        // defaults to cubic-ease-in.
-        velocityFn:( velocity:number, threshold:number ) => number;
+        // is already at left/top edge
+        return (currentScrollOffset <= 0);
     }
+    // no scroll
+    return true;
+}
 
-    export function SetOptions( options:ScrollOptions ):void {
+//</editor-fold>
 
-        // overwrite defaults with input options
-        Object.keys( options ).forEach( function( key ) {
-            _options[ key ] = options[ key ];
-        } );
-    }
+//<editor-fold desc="public api">
 
-    export const HandleDragImageTranslateOverride:DragImageTranslateOverrideFn = handleDragImageTranslateOverride;
+export interface ScrollOptions {
+    // threshold in px. when distance between scrollable element edge and touch position is smaller start programmatic scroll.
+    // defaults to 75px
+    threshold?:number;
+    // function to customize the scroll velocity
+    // velocity param: distance to scrollable element edge
+    // threshold: the threshold used to determine when scrolling should start
+    // defaults to cubic-ease-in.
+    velocityFn:( velocity:number, threshold:number ) => number;
+}
+
+function SetOptions( options:ScrollOptions ):void {
 
-    //</editor-fold>
+    // overwrite defaults with input options
+    Object.keys( options ).forEach( function( key ) {
+        _options[ key ] = options[ key ];
+    } );
 }
+
+export const HandleDragImageTranslateOverride:DragImageTranslateOverrideFn = handleDragImageTranslateOverride;
```

Comment:
Maybe this should get a specific "scroll" related name to avoid conflicts on the global "DragDropPolyfill" namespace.